src/assets/scripts/client/trafficGenerator/SpawnPatternModel.js

```diff
@@ -995,19 +996,30 @@ export default class SpawnPatternModel extends BaseModel {
             return [initialWaypoint, nextWaypoint];
         }
 
+        const isPreSpawn = false;
         const routeModel = new RouteModel(formattedRoute[0]);
-        const waypointModelList = navigationLibrary.findEntryAndBodyFixesForRoute(
+        const waypointModelList = navigationLibrary.starCollection.findRouteWaypointsForRouteByEntryAndExit(
             routeModel.procedure,
-            routeModel.entry
+            routeModel.entry,
+            this._airportController.getInitialArrivalRunwayName(),
+            isPreSpawn
         );
 
         return waypointModelList;
     }
 
+    /**
+     *
+     * @for SpawnPatternModel
+     * @method _generateSelfReferencedAirportPositionModel
+     * @return {StaticPositionModel}
+     */
     _generateSelfReferencedAirportPositionModel() {
```

Comment:
Airports are the center of the world. So they don't have a "reference position" (which is basically used for the rotation from magvar).